### PR TITLE
Add rcparam for figure label size and weight

### LINKED ISSUE #22564: [ENH]: Font weight for Figure title and labels
### Problem

Figure title (_Figure.suptitle()_) and labels (_Figure.supxlabel()_ and _Figure.supylabel()_) have the same default font size and weight. Although one can modify these values when creating the title and/or labels, it would be easier if users were able to set different defaults.

### Proposed solution

Create rcParams called _figure.labelweight_ and _figure.labelsize_, so that _Figure.supxlabel()_ and _Figure.supylabel()_ would default to these new values, rather than _rcParams["figure.titleweight"]_ and _rcParams["figure.titlesize"]_.

### PR DESCRIPTION
**Pull request by oscargus**
## PR Summary

Closes  #22564

Add `figure.labelsize` and `figure.labelweight` rcParams to set figure label size and weight different from title.

Test added also for titlesize/-weight rcParams as I didn't find any.

## PR Checklist

**Tests and Styling**
- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [N/A] New features are documented, with examples if plot related.
- [x] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [N/A] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

File: lib/matplotlib/figure.py [hunk 1]
```diff
@@ -331,8 +331,8 @@ def _suplabels(self, t, info, **kwargs):
         fontproperties : None or dict, optional
             A dict of font properties. If *fontproperties* is given the
             default values for font size and weight are taken from the
```
Comment (oscargus) [inline lib/matplotlib/figure.py:333]:
```suggestion
            default, values for font size and weight are taken from the
```

File: doc/users/next_whats_new/figure_label_rcparams.rst [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Allow setting figure label size and weight globally and separately from title
+-----------------------------------------------------------------------------
+
+The figure label, ``Figure.supxlabel`` and ``Figure.supylabel``, size and weight
```
Comment (oscargus) [inline doc/users/next_whats_new/figure_label_rcparams.rst:4]:
```suggestion
The figure labels, ``Figure.supxlabel`` and ``Figure.supylabel``, size and weight
```

**Review by jklymak — COMMENTED** [review]
This seems good.  Needs a rebase, so I'll pop to draft until that is done.  You also self-review, but don't seem to have merged your suggestions yet....

**Comment by oscargus** [comment]
Thanks! I'll try to sort it out. I thought there were more to do here (related to a discussion in another PR about how to deal with default values, but I clearly forgot about the status as well).

**Comment by oscargus** [comment]
Rebased.

File: lib/matplotlib/figure.py (hunk 1 above)
Comment (oscargus) [reply lib/matplotlib/figure.py:333]: Maybe `If *fontproperties* is not given, the ...`?

Or maybe it is correct, just that I have problem understanding?

**Review by jklymak — APPROVED** [review]

File: lib/matplotlib/tests/test_figure.py [hunk 3]
```diff
@@ -1370,6 +1370,20 @@ def test_kwargs_pass():
...
+    fig_ref.supxlabel("xlabel", weight='bold', size=15)
+    fig_ref.supylabel("ylabel", weight='bold', size=15)
+    fig_ref.suptitle("Title", weight='light', size=20)
+    with mpl.rc_context({'figure.labelweight': 'bold',
+                         'figure.labelsize': 15,
+                         'figure.titleweight': 'light',
+                         'figure.titlesize': 20}):
+        fig_test.supxlabel("xlabel")
+        fig_test.supylabel("ylabel")
+        fig_test.suptitle("Title", weight='light')
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_figure.py:1384]: Shouldn't the `weight` not be here?

Comment (oscargus) [reply lib/matplotlib/tests/test_figure.py:1384]: Indeed! Well spotted!

**Comment by oscargus** [comment]
Please squash-merge. I have a bit of an issue with a rebase setting in my new git-install, so I do not dare to touch it in some situations...

**Pull request merged**